src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,50 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final String WORDS_SEPARATOR = " - ";
+    private static final String SYSTEM_SEPARATOR = System.lineSeparator();
+
+    public String getSalaryInfo(String[] names, String[] data,
+                                       String dateFrom, String dateTo) {
+        int[] salaryCount = new int[names.length];
+        for (String datum : data) {
+            String[] dataAboutEmployee = datum.split(" ");
+            if (dataValidation(dataAboutEmployee[0], dateFrom, dateTo)) {
+                for (int i = 0; i < names.length; i++) {
+                    if (names[i].equals(dataAboutEmployee[1])) {
+                        salaryCount[i] += Integer.parseInt(dataAboutEmployee[2])
+                                * Integer.parseInt(dataAboutEmployee[3]);
+                    }
+                }
+            }
+        }
+        return getReport(names, salaryCount, dateFrom, dateTo);
+    }
+
+    private boolean dataValidation(String date, String dateFrom, String dateTo) {
+        String datePattern = "dd.MM.yyyy";
+        SimpleDateFormat dateFormat = new SimpleDateFormat(datePattern);
+        try {
+            if (dateFormat.parse(date).compareTo(dateFormat.parse(dateFrom)) >= 0
+                             && dateFormat.parse(date).compareTo(dateFormat.parse(dateTo)) <= 0) {
+                return true;
+            }
+        } catch (ParseException e) {
+            throw new RuntimeException("Invalid input date format", e);
+        }
+        return false;
+    }
+
+    private String getReport(String[] names, int[] salary, String dateFrom, String dateTo) {
```

Comment:
if we do not use this method, we can get rid of one cycle if we store data not in int[] salaryCount but in StringBuilder